packages/transmute-framework/src/stream-model/__tests__/sync.spec.js

```diff
@@ -24,80 +43,74 @@ describe('sync', () => {
     eventStore = await eventStore.clone(accounts[0]);
   });
 
-  it('handles the empty case', async () => {
-    let newEventStore = await eventStore.clone(accounts[0]);
-    const filter = event => {
-      return true;
-    };
-    const reducer = (state, event) => {
-      return state;
-    };
-    const streamModel = new StreamModel(newEventStore, filter, reducer);
+  it('should handle the empty case, where lastIndex is always null', async () => {
+    const streamModel = new StreamModel(eventStore, filter, reducer);
+    // sync with no events in contract
     await streamModel.sync();
     expect(streamModel.state.lastIndex).toBe(null);
   });
 
-  it('handles new events', async () => {
-    let newEventStore = await eventStore.clone(accounts[0]);
-    const filter = event => {
-      return true;
-    };
-    const reducer = (state, event) => {
-      return {
-        ...state,
-        eventsProcessed: (state.eventsProcessed | 0) + 1
-      };
-    };
-    const streamModel = new StreamModel(newEventStore, filter, reducer);
-    await newEventStore.write(
-      accounts[0],
-      mockEvents[0].key,
-      mockEvents[0].value
-    );
+  it('should handle new events, where lastIndex is the id of the last event', async () => {
+    const streamModel = new StreamModel(eventStore, filter, reducer);
+
+    await eventStore.write(accounts[0], mockEvents[0].key, mockEvents[0].value);
+
     await streamModel.sync();
     expect(streamModel.state.model.eventsProcessed).toBe(1);
-  });
+    expect(streamModel.state.lastIndex).toBe(0);
 
-  it('handles persisted state', async () => {
-    let newEventStore = await eventStore.clone(accounts[0]);
-    const filter = event => {
-      return true;
-    };
-    const reducer = (state, event) => {
-      let eventHash = newEventStore.web3.sha3(JSON.stringify(event));
-      const eventHashes = new Set(state.eventHashes || []);
-      eventHashes.add(eventHash);
-      return {
-        ...state,
-        eventHashes: Array.from(eventHashes)
-      };
-    };
-    const streamModel = new StreamModel(newEventStore, filter, reducer);
-    await newEventStore.write(
-      accounts[0],
-      mockEvents[0].key,
-      mockEvents[0].value
-    );
     await streamModel.sync();
+    expect(streamModel.state.model.eventsProcessed).toBe(1);
     expect(streamModel.state.lastIndex).toBe(0);
 
-    await newEventStore.write(
-      accounts[0],
-      mockEvents[1].key,
-      mockEvents[1].value
-    );
+    await eventStore.write(accounts[0], mockEvents[1].key, mockEvents[1].value);
+
+    await streamModel.sync();
+    expect(streamModel.state.model.eventsProcessed).toBe(2);
+    expect(streamModel.state.lastIndex).toBe(1);
+  });
 
-    // console.log(JSON.stringify(streamModel.state, null, 2));
+  it('should handle persisted state', async () => {
+    const streamModel = new StreamModel(eventStore, filter, reducer);
```

Comment:
Move the declaration of `streamModel` in `beforeAll()`. It is declared in every test with the same parameters

Reply:
agree.